R/monitor.R

```diff
@@ -0,0 +1,151 @@
+#' over time
+#' - `vetiver_plot_metrics()` creates a plot of metrics over time
+#'
+#' @inheritParams yardstick::metrics
+#' @inheritParams pins::pin_read
+#' @inheritParams slider::slide_period
+#' @param date_var The column in `data` containing dates or date-times for
+#' monitoring, to be aggregated with `unit`
+#' @param metric_set A [yardstick::metric_set()] function for computing metrics.
+#' Defaults to [yardstick::metrics()].
+#' @param df_metrics A tidy dataframe of metrics over time, such as created by
+#' `vetiver_compute_metrics()`.
+#' @param metrics_pin_name Pin name for where the *metrics* are stored (as
+#' opposed to where the model object is stored with [vetiver_pin_write()]).
+#' @param initiate Are you starting a new set of monitoring metrics to
+#' pin/store? Defaults to `FALSE`.
+#' @param .estimate The variable in `df_metrics` containing the metric estimate.
+#' Defaults to `.estimate`.
+#' @param .metric The variable in `df_metrics` containing the metric type.
+#' Defaults to `.metric`.
+#' @param n The variable in `df_metrics` containing the number of observations
+#' used for estimating the metric.
+#'
+#' @return Both `vetiver_compute_metrics()` and `vetiver_pin_metrics()` return
+#' a dataframe of metrics. The `vetiver_plot_metrics()` function returns a
+#' `ggplot2` object.
+#'
+#' @details For arguments used more than once in your monitoring dashboard,
+#' such as `date_var`, consider using
+#' [R Markdown parameters](https://bookdown.org/yihui/rmarkdown/parameterized-reports.html)
+#' to reduce repetition and/or errors.
+#'
+#' @examples
+#' library(dplyr)
+#' library(parsnip)
+#' data(Chicago, package = "modeldata")
+#' Chicago <- Chicago %>% select(ridership, date, one_of(stations))
+#' training_data <- Chicago %>% filter(date < "2009-01-01")
+#' testing_data <- Chicago %>% filter(date >= "2009-01-01", date < "2011-01-01")
+#' monitoring <- Chicago %>% filter(date >= "2011-01-01", date < "2012-12-31")
+#' lm_fit <- linear_reg() %>% fit(ridership ~ ., data = training_data)
+#'
+#' library(pins)
+#' b <- board_temp()
+#'
+#' ## before starting monitoring, initiate the metrics and pin
+#' ## (for example, with the testing data):
+#' original_metrics <-
+#'     augment(lm_fit, new_data = testing_data) %>%
+#'     vetiver_compute_metrics(date, "week", ridership, .pred, .every = 4L) %>%
+#'     vetiver_pin_metrics(date, b, "lm_fit_metrics", initiate = TRUE)
+#'
+#' ## to continue monitoring with new data, compute metrics and update pin:
+#' new_metrics <-
+#'     augment(lm_fit, new_data = monitoring) %>%
+#'     vetiver_compute_metrics(date, "week", ridership, .pred, .every = 4L) %>%
+#'     vetiver_pin_metrics(date, b, "lm_fit_metrics")
+#'
+#' library(ggplot2)
+#' vetiver_plot_metrics(new_metrics, date) +
+#'     scale_size(range = c(2, 4))
+#'
+#' @export
+vetiver_compute_metrics <- function(data,
+                                    date_var,
+                                    .period,
```

Comment:
Addressed in rewrite.

I don't think you actually need to put a `.` in front of the slider args. This was only done because the `...` in `slide_period_dfr()` could be named and we want to avoid conflicts with what the user could put in. But you control what is passed through here, and there are no conflicts.

Reply:
I really wanted to inherit the docs for parameters from slider so I don't accidentally get out of sync. (This has happened in other packages.) Do you know of a way to inherit arg docs that have a different name, like using `.period` docs for `period`?

Reply:
It works AUTOMATICALLY for that specific case 🔥